### PR TITLE
Ians Fork of: Finish up SDK caching #440

### PR DESCRIPTION
**Pull request by ianblenke**
Based on #440, this holds the changes necessary for, per @shareefalis:
- zack was close to finishing up the caching step but he attempted to caching the wrong directory.
- So finish that up for him and it works!

### CODE REVIEW

File: Tools/travis_compile.sh [hunk 1]
```diff
@@ -3,6 +3,27 @@ set -xe
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 cd $DIR/..
 
+mkdir -p sdkcache/
+git submodule | cut -d'(' -f1 | sort > sdkcache/current.txt
```
Comment (shareefalis) [inline Tools/travis_compile.sh:7]: git submodule status  --recursive so you can get all the submodule in  submodules

File: cachezip.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
```
Comment (shareefalis) [inline cachezip.sh]: you can probably remove this also

File: Tools/travis_compile.sh (hunk 1 above)
Comment (ianblenke) [inline Tools/travis_compile.sh:7]: We are only concerned with the SHA1 of the actual submodules in the repo, not the submodules in the submodules. The reasoning is that if you update a submodule of a submodule, when you go to git-add/git-commit it, that will result in a new SHA1 that must in turn be added/committed to the repo, which will show as a different SHA1... meaning, recursion is irrelevant for the purposes of knowing when submodules have changed.

Comment (shareefalis) [inline Tools/travis_compile.sh:7]: Good point, carry away.

**Comment by zackm0571** [comment]
+1

**Pull request merged**